src/renderer/src/components/settings-modal/audio-apis.tsx

```diff
@@ -8,13 +8,25 @@ export interface AudioApisProps {
 }
 
 const AudioApis: React.FC<AudioApisProps> = ({ apis, selectedApiId, selectApi }) => {
+  let apiValue = -1;
+
+  // Help the user out a bit by defaulting the audio API to the only valid
+  // entry if no API was selected and the list of available APIs only
+  // has one entry.
+  if (apis.some((api) => api.id === selectedApiId)) {
```

Comment:
All of this logic needs to be wrapped un useEffect, apiValue should be a useState

Reply:
Does it? It only has to happen the very first time the dialog is created, and it's not used as a variable to trigger redraws/refreshes of anything in the UI.

Reply:
It cannot be guaranteed that this will only be executed once if not in a useEffect wrap, that's why